index.js

```diff
@@ -38,14 +41,31 @@ function loadModule(config) {
  */
 function initializeBookend(defaultModules, list) {
     return list.map((m) => {
-        if (typeof m === 'string' && defaultModules[m]) {
+        let name;
+        let alias;
+
+        if (typeof m === 'string') {
+            name = m;
+            alias = m;
+        } else {
+            // Use the first key as a module name if name is not set
+            // to support the format: { moduleName: moduleAlias }
+            name = m.name || Object.keys(m)[0];
+            alias = m[name] || name;
```

Comment:
I've got to say that I found this alias configuration really confusing. Personally, I'd rather explicitly call out the properties in the configuration object, than make them "magical" like these are.

Something like this is very concise and leaves no room for confusion:
```
const b = new Bookend(defaultModules, [], [
    { name: 'greeting', alias: 'foo' },
    { name: 'sample', alias: 'bar' },
    { name: 'planet', alias: 'baz' }
]);
```

Reply:
If you really wanted to maintain the simplicity of the string based config, you could still do that, but it better be REALLY well documented:
```
// Split name:alias at the ':'
const b = new Bookend(defaultModules, [
    'greeting:foo',
    'sample:bar',
    'planet:baz'
], []);
```

Reply:
@petey I agree with you. It looks simple to implement and use. I'll fix to accept this schema.